private/bufpkg/bufapi/bufapi.go

```diff
@@ -37,6 +37,13 @@ func NewClientProvider(clientConfig *connectclient.Config) ClientProvider {
 	return newClientProvider(clientConfig)
 }
 
+// NewNopClientProvider returns a new ClientProvider that provides unimplemented services.
+//
+// Should be used for testing only.
+func NewNopClientProvider() ClientProvider {
+	return newNopClientProvider()
```

Comment:
Nit: In other places, we just do `var NopClientProvider ClientProvider = nopClientProvider{}` as a global variable, and then you don't have any constructors, and you implement the functions directly on the struct, not the struct pointer, ie `func (nopClientProvider) BranchServiceClient...`